toolkit/tools/internal/pkggraph/pkggraph_test.go

```diff
@@ -949,6 +949,13 @@ func TestReadWriteGraph(t *testing.T) {
 
 // Validate the reference graph is valid, and that it matches the output of the test graph.
 func TestReferenceDOTFile(t *testing.T) {
+	if testing.Short() {
+		// Encoding is unreliable on some systems. The final output is still a valid
+		// graph but the base64 encoding of the nodes is different. The final graph once
+		// decoded is the same however (probably gob encoding is different since its stateful?)
```

Comment:
If the point of this serialization is to produce an opaque blob we can deserialize back and get semantically equivalent results -- then shouldn't the test just round-trip the graphs and compare results instead of doing anything with binary representation? Sounds like an overly constrained test contract.

Reply:
I'm leaning towards this as well, but this has been reliable for ~4 years so I'm curious why it has changed all of a sudden.